### PR TITLE
Update to support OpenSSL 1.1

### LINKED ISSUE #34: support 1.1
a few methods are not supported. e.g. `OPENSSL_init_ssl`

### PR DESCRIPTION
**Pull request by jeking3**
add support for "static if (OPENSSL_VERSION_AT_LEAST(1, 1))"
add support for "static if (OPENSSL_VERSION_BEFORE(1, 1))"
those changes are in opensslv.d and they are forward-facing

if you want to keep your code compatible with OpenSSL 1.0 and 1.1 you can do something like this for now since OPENSSL_VERSION_{AT_LEAST|BEFORE} won't exist in previous releases so you cannot use them yet:

>     static if (OPENSSL_VERSION_NUMBER >= 0x1010000f) { // OPENSSL_VERSION_AT_LEAST(1, 1)) {
>         ctx_ = SSL_CTX_new(TLS_method());
>     } else {
>         ctx_ = SSL_CTX_new(SSLv23_method());

then once you are using 1.1 code you can switch to the macro for future static branches.
I also fixed up the stack.d conditional to use that instead of checking for version(DeimosOpenSSLv1_1).

This fixes #34

### CODE REVIEW

**Comment by jeking3** [comment]
This has allowed the Apache Thrift docker build using Ubuntu Bionic (which comes with 1.1) to build and pass all tests.

**Comment by jeking3** [comment]
After some additional testing, this causes undefined symbol messages on openssl-1.0 links for a few new functions.

File: deimos/openssl/stack.d [hunk 1]
```diff
@@ -89,7 +90,7 @@ char* M_sk_value()(_STACK* sk, size_t n) { return (sk ? sk.data[n] : null); }
 
 *******************************************************************************/
 
-version (DeimosOpenSSLv1_1)
+static if (OPENSSL_VERSION_AT_LEAST(1, 1))
```
Comment (jeking3) [inline deimos/openssl/stack.d]: While it does no harm, there's perhaps no reason to have any backwards compatibility stuff in here (unless it's going to be backported to support openssl-1.0, but there's already a release for that).

**Comment by MartinNowak** [comment]
New release still needs to be tagged as `2.0.0+1.1.0h`.
As the header is no longer compatible with OpenSSL 1.0, I'd say a new major semver version is due.

**Comment by CyberShadow** [comment]
@MartinNowak Some existing discussion here: https://github.com/D-Programming-Deimos/openssl/issues/34

> As the header is no longer compatible with OpenSSL 1.0,

That's a problem. Is there any way for D programs to be buildable/runnable on both systems with OpenSSL 1.0 and 1.1? (E.g. Ubuntu LTS and Arch Linux)

**Comment by jeking3** [comment]
Buildable - yes, but you need to specify the version (~>1.1.6) or (~>2.0.0) of the deimos corresponding with the OpenSSL in your dependencies.  I agree it would be better if it "just worked" with either underlying installed library version.

**Pull request merged**